src/Microsoft.Health.Fhir.CosmosDb/Features/Storage/FhirCosmosClientInitializer.cs

```diff
@@ -43,45 +48,38 @@ public class FhirCosmosClientInitializer : ICosmosClientInitializer
             _requestHandlerFactory = requestHandlerFactory;
             _retryExceptionPolicyFactory = retryExceptionPolicyFactory;
             _logger = logger;
+            _cosmosClients = new ConcurrentDictionary<string, CosmosClient>();
+            _lockObject = new object();
         }
 
         /// <inheritdoc />
         public CosmosClient CreateCosmosClient(CosmosDataStoreConfiguration configuration)
         {
             EnsureArg.IsNotNull(configuration, nameof(configuration));
 
-            var host = configuration.Host;
-            var key = configuration.Key;
+            string key = string.IsNullOrEmpty(configuration.Host) ? "local" : configuration.Host;
```

Comment:
qq: what was the reason to need multiple clients?

Reply:
I believe this part of the code can be simpler: a single CosmosClient would be appropriate.

Reply:
Done. Added simple thread safe logic to create CosmosClient